### PR TITLE
fix(reviewer): align prompt format with parser and improve fallback handling

### LINKED ISSUE #407: Bug: task_reviewer prompt/parser mismatch causes workflows to abort without completing all tasks
## Summary

Workflows using the `task_reviewer` agent (haiku model) get stuck retrying a single task and never advance to subsequent tasks. After exhausting the max iteration limit, the workflow aborts entirely instead of completing the remaining tasks.

Observed in workflow **TEST-001** (TASK-001 on `/Users/ka/github/existential-birds/jocko`): A 5-task plan repeated "Starting Task 1/5" five times, the reviewer never approved, and the workflow ended marked as "completed" with only Task 1 attempted.

## Root Cause

There are two compounding issues:

### 1. Prompt instructs JSON output, but parser expects markdown

**Server mode** resolves the reviewer prompt from `PROMPT_DEFAULTS["reviewer.agentic"]` in `amelia/agents/prompts/defaults.py`, which instructs the model to output **JSON**:

```json
{"approved": true|false, "comments": [...], "severity": "..."}
```

But `_parse_review_result()` in `amelia/agents/reviewer.py` (lines 350-498) parses **markdown**, looking for `Ready: Yes|No` via regex:

```python
verdict_match = re.search(
    r"[*_]{0,2}Ready[*_]{0,2}:[*_]{0,2}\s*(Yes|No|With fixes[^\n]*)",
    output,
    re.IGNORECASE,
)
```

The parser has **zero JSON awareness**. When the model follows the JSON prompt instructions, the primary regex always fails.

The prompt resolution path is:
- `Reviewer.agentic_prompt` property (line 128-130) checks `self._prompts.get("reviewer.agentic", self.AGENTIC_REVIEW_PROMPT)`
- In server mode, `PromptResolver` populates prompts from `PROMPT_DEFAULTS` (JSON format) → the JSON prompt is used
- In CLI mode (no prompts dict), the fallback `AGENTIC_REVIEW_PROMPT` class constant (markdown format) is used → the parser works correctly

### 2. Fallback keyword scanner is fragile

When the primary regex fails, `_parse_review_result()` falls back to keyword scanning (lines 404-424):

```python
approval_keywords_found = [
    word for word in ["ready: yes", "approved", "lgtm", "looks good"]
    if word in output_lower
]
rejection_keywords_found = [
    word for word in ["ready: no", "not approved", "needs fixes", "blocked"]
    if word in output_lower
]
```

This is unreliable because:
- In JSON output, `"approved"` appears as a key in both `"approved": true` AND `"approved": false`
- Any commentary containing `"needs fixes"` overrides approval (rejection keywords take precedence)
- Haiku as a smaller model is especially prone to format deviations and adding commentary

### 3. Max iterations aborts the entire workflow instead of advancing

In `route_after_task_review()` (`amelia/pipelines/implementation/routing.py`, lines 43-104):

```python
if state.task_review_iteration >= max_iterations:
    return "__end__"  # Halt on repeated failure
```

When the reviewer never approves (due to the parsing bug), the workflow exhausts all retry attempts on a single task and then **terminates the entire workflow** — never reaching `next_task_node` to advance `current_task_index`.

## Reproduction

The failure path:
1. Start a workflow via the server with a multi-task plan
2. The `task_reviewer` agent (typically haiku) runs via `call_reviewer_node()`
3. The prompt from `defaults.py` tells haiku to output JSON
4. `_parse_review_result()` fails the primary regex (no `Ready:` pattern in JSON)
5. Fallback keyword scanning produces `approved=False`
6. `route_after_task_review()` routes back to `developer` (retry)
7. After 5 iterations (default `max_iterations`), routing returns `"__end__"`
8. Workflow ends without completing tasks 2-5

## Affected Files

| File | Role |
|------|------|
| `amelia/agents/prompts/defaults.py` | `PROMPT_DEFAULTS["reviewer.agentic"]` — instructs JSON output |
| `amelia/agents/reviewer.py` | `AGENTIC_REVIEW_PROMPT` — instructs markdown output; `_parse_review_result()` — parses markdown only |
| `amelia/pipelines/implementation/routing.py` | `route_after_task_review()` — aborts on max iterations |
| `amelia/pipelines/implementation/nodes.py` | `next_task_node()` — never reached due to routing |
| `amelia/pipelines/nodes.py` | `call_reviewer_node()` — selects `task_reviewer` config for non-final tasks |

## Test Coverage Gaps

`TestParseReviewResult` in `tests/unit/agents/test_reviewer.py` is missing:
- JSON-format output (`{"approved": true, ...}`)
- Output that omits the `## Verdict` section entirely
- Alternative phrasings (`Verdict: Approved`, `Status: Pass`)
- Truncated/incomplete output where the verdict is cut off
- Haiku-style terse output without markdown structure

## Suggested Fixes

1. **Align prompt and parser**: Either update `defaults.py` to use the markdown `Ready: Yes|No` format, or add JSON parsing to `_parse_review_result()`
2. **Advance on max iterations instead of aborting**: Change `route_after_task_review()` to route to `next_task_node` (with a warning event) when max iterations are exhausted on a non-final task, rather than returning `__end__`
3. **Add test coverage**: Cover JSON output, alternative phrasings, and edge cases in `TestParseReviewResult`

### PR DESCRIPTION
**Pull request by anderskev**
## Summary

Fixes three compounding bugs in the task reviewer that caused infinite developer-reviewer loops and premature workflow termination (#407). The reviewer prompt instructed JSON output while the parser expected markdown with `Ready: Yes|No`, the keyword fallback produced false positives, and max-iterations aborted the entire workflow instead of advancing.

### Fixed
- **Prompt/parser mismatch**: Extracted `REVIEW_OUTPUT_FORMAT` constant from the parser and injected it into both `Reviewer.AGENTIC_REVIEW_PROMPT` and `PROMPT_DEFAULTS['reviewer.agentic']`, so the LLM output format stays in sync with what `_parse_review_result()` expects
- **Fragile keyword fallback**: Replaced the keyword scanner (which matched "approved" regardless of context) with an issue-aware heuristic — no verdict + issues found = reject, no verdict + no issues = approve
- **Max-iterations routing**: `route_after_task_review` now advances to the next task on max iterations for non-final tasks instead of aborting the entire multi-task workflow
- **Verdict regex**: Replaced `[*_]{0,2}` with `[*_]*` to handle any combination of markdown bold/italic formatting (e.g., `***Ready***:`)
- **Log level**: Downgraded fallback approval log from info to debug (expected behavior, not noteworthy)

### Changed
- Extracted `MAX_REVIEW_COMMENTS` constant (was magic number `20`) with docstring explaining its purpose

### Refactored
- Tests now use Enum types (`DriverType.CLI`, `TrackerType.NOOP`, `Severity.NONE`) instead of string literals for type safety

## Motivation

Issue #407: The reviewer agent prompt instructed the LLM to output JSON (`{"approved": true, ...}`) while `_parse_review_result()` expected structured markdown with `Ready: Yes|No`. This caused the verdict regex to never match, falling through to a keyword scanner that would false-positive on words like "approved" appearing in prose. When the reviewer kept rejecting, max-iterations would terminate the entire workflow rather than advancing to the next task.

## Testing

- [x] Unit tests added/updated
- [x] Integration tests added/updated
- [ ] Manual testing performed

### Test Coverage

- `test_reviewer.py`: `_parse_review_result` correctly defaults to `approved=False` when no Ready: verdict found (no keyword fallback); issue-aware heuristic approves when no issues present
- `test_reviewer_prompts.py`: Verifies prompt contains `REVIEW_OUTPUT_FORMAT` sections
- `test_routing.py`: All branches of `route_after_task_review` — approved final/non-final, not-approved within iterations, max-iterations on final vs non-final task
- `test_reviewer_prompt_parser.py`: Integration test verifying the full chain: prompt template → driver call → parse → ReviewResult → routing

## Related Issues

- Closes #407

## Checklist

- [x] Code follows project style guidelines
- [x] Self-review completed
- [x] Tests pass locally
- [x] Linting passes
- [x] Documentation updated (design doc included)

---

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/plans/2026-02-08-task-reviewer-prompt-parser-fix-design.md [hunk 1]
```diff
@@ -0,0 +1,837 @@
...
+
+**Step 2: Run linting and type checking**
+
+Run: `uv run ruff check amelia tests`
+Run: `uv run mypy amelia`
+Expected: No errors
+
+**Step 3: Verify no regressions in existing reviewer tests**
+
+Run: `uv run pytest tests/unit/agents/test_reviewer.py tests/unit/agents/test_reviewer_prompts.py tests/unit/pipelines/test_routing.py -v`
```
[the remark this answers is not in the document]
Comment (anderskev) [reply docs/plans/2026-02-08-task-reviewer-prompt-parser-fix-design.md]: @coderabbitai The design doc was deleted entirely in 035746c3 after implementation completed. No action needed.

File: tests/integration/test_reviewer_prompt_parser.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            status="running",
+            current_task_index=0,
+            total_tasks=3,
+            last_review=result,
+        )
+        route = route_after_task_review(state_after, profile)
+        # Approved non-final task -> advance to next task
+        assert route == "next_task_node"
+
+    async def test_prompt_defaults_contain_ready_format(self) -> None:
```
[the remark this answers is not in the document]
Comment (anderskev) [reply tests/integration/test_reviewer_prompt_parser.py]: @coderabbitai This project uses `asyncio_mode = "auto"` in pyproject.toml, which makes the explicit `@pytest.mark.asyncio` marker unnecessary — pytest-asyncio auto-detects async test functions. No integration test in this repo uses the marker (0/20 files). Won't fix.
